ouroboros-consensus-cardano-tools/src/Cardano/Node/Protocol/Conway.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE EmptyDataDeriving #-}
+
+-- TODO DUPLICATE? -- as-if adapted? from: cardano-node/src/Cardano/Node/Protocol/Conway.hs
+
+module Cardano.Node.Protocol.Conway (
+    ConwayProtocolInstantiationError
+    -- * Reusable parts
+  , readGenesis
+  , validateGenesis
+  ) where
+
+import           Cardano.Prelude
+
+import qualified Cardano.Ledger.Conway.Genesis as Conway
+import qualified Cardano.Ledger.Crypto as Crypto
+
+import           Cardano.Node.Protocol.Shelley (GenesisReadError,
+                     readGenesisAny)
+import           Cardano.Node.Types
+
+--
+-- Conway genesis
+--
+
+readGenesis :: Crypto.Crypto c
+            => GenesisFile
+            -> Maybe GenesisHash
+            -> ExceptT GenesisReadError IO
+                       (Conway.ConwayGenesis c, GenesisHash)
+readGenesis = readGenesisAny
+
+validateGenesis :: Conway.ConwayGenesis c
+                -> ExceptT ConwayProtocolInstantiationError IO ()
+validateGenesis _ = return () --TODO conway: do the validation
```

Comment:
Should we leave this `TODO` and the ones below? This is only used by the `db-synthesizer`, so implementing this is out of scope. Also note that [this is not implemented for Alonzo either](https://github.com/input-output-hk/ouroboros-network/issues/4066).

Reply:
> Also note that this is not implemented for Alonzo either.

😬 

TODO actually, see my other comment. I'm wondering if there is no genesis data for Conway now. So we wouldn't need this module (or many of this PR's other changes in the `o-c-c-tools` package).

Reply:
There is still Conway Genesis data, so no simple option there :(

I think this remains a TODO (as for Alonzo, as you noted) until the genesis/genesis file validation logic is available for use as a library. Two options:

- that logic is already available in Ledger code, we just need to find it, import it, call it.

- that logic is only available in Node code, in which case we have to deal with the awkward upside-downness of our dependencies and repos (eg upstream that validation here (in to a tool?!) so that both db-analyser and -node could use it :/ )